conans/client/rest/conan_requester.py

```diff
@@ -0,0 +1,69 @@
+import fnmatch
+import os
+
+from conans.util.files import save
+
+
+class ConanRequester(object):
+
+    def __init__(self, requester, client_cache):
+        self.proxies = client_cache.conan_config.proxies or {}
+        self.no_proxy_match = [el.strip() for el in
+                               self.proxies.pop("no_proxy_match", "").split(",")]
+
+        # Retrocompatibility with deprecated no_proxy
+        # Account for the requests NO_PROXY env variable, not defined as a proxy like http=
+        no_proxy = self.proxies.pop("no_proxy", None)
+        if no_proxy:
+            os.environ["NO_PROXY"] = no_proxy
+
+        self._requester = requester
+        self._client_cache = client_cache
+
+        if not os.path.exists(client_cache.client_cert_path):
+            self._client_certificates = None
+        else:
+            if os.path.exists(client_cache.client_cert_key_path):
+                # Requests can accept a tuple with cert and key, or just an string with a
+                # file having both
+                self._client_certificates = (client_cache.client_cert_path,
+                                             client_cache.client_cert_key_path)
+            else:
+                self._client_certificates = client_cache.client_cert_path
+
+    def _should_skip_proxy(self, url):
+        if not self.no_proxy_match:
+            return False
+
+        for entry in self.no_proxy_match:
+            if fnmatch.fnmatch(url, entry):
+                return True
+
+        return False
+
+    def _add_kwargs(self, url, kwargs):
+        if kwargs.get("verify", None) is True:
+            if not os.path.exists(self._client_cache.cacert_path):
```

Comment:
A bit ugly in this method, called all the time? Maybe the path once in the constructor? Maybe created by ``client_cache``?